src/NClient/NClient.Standalone/ClientProxy/Generation/MethodBuilders/Providers/TimeoutAttributeProvider.cs

```diff
@@ -0,0 +1,54 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using NClient.Annotations;
+using NClient.Core.Mappers;
+using NClient.Standalone.Exceptions.Factories;
+
+namespace NClient.Standalone.ClientProxy.Generation.MethodBuilders.Providers
+{
+    internal interface ITimeoutAttributeProvider
+    {
+        ITimeoutAttribute? Get(MethodInfo method, IEnumerable<MethodInfo> overridingMethods);
+    }
+
+    internal class TimeoutAttributeProvider : ITimeoutAttributeProvider
+    {
+        private readonly IAttributeMapper _attributeMapper;
+        private readonly IClientValidationExceptionFactory _clientValidationExceptionFactory;
+
+        public TimeoutAttributeProvider(
+            IAttributeMapper attributeMapper,
+            IClientValidationExceptionFactory clientValidationExceptionFactory)
+        {
+            _attributeMapper = attributeMapper;
+            _clientValidationExceptionFactory = clientValidationExceptionFactory;
+        }
+
+        public ITimeoutAttribute? Get(MethodInfo method, IEnumerable<MethodInfo> overridingMethods)
+        {
+            return Find(method) ?? overridingMethods.Select(Find).FirstOrDefault();
+        }
+
+        private TimeoutAttribute? Find(MethodInfo method)
+        {
+            var attributes = method
+                .GetCustomAttributes()
+                .Select(x => _attributeMapper.TryMap(x))
+                .ToArray();
+            if (attributes.Any(x => x is IPathAttribute))
+                throw _clientValidationExceptionFactory.MethodAttributeNotSupported(nameof(IPathAttribute));
```

Comment:
This is not required here, the paths are checked in another provider

Reply:
Yep. Removed